receiver/sqlserverreceiver/scraper.go

```diff
@@ -1,144 +1,147 @@
 // Copyright The OpenTelemetry Authors
 // SPDX-License-Identifier: Apache-2.0
 
-//go:build windows
-
 package sqlserverreceiver // import "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/sqlserverreceiver"
 
 import (
 	"context"
+	"database/sql"
+	"errors"
+	"fmt"
+	"strconv"
 	"time"
 
 	"go.opentelemetry.io/collector/component"
 	"go.opentelemetry.io/collector/pdata/pcommon"
 	"go.opentelemetry.io/collector/pdata/pmetric"
-	"go.opentelemetry.io/collector/receiver"
+	"go.opentelemetry.io/collector/receiver/scraperhelper"
 	"go.uber.org/multierr"
 	"go.uber.org/zap"
 
-	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/winperfcounters"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/internal/sqlquery"
 	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/sqlserverreceiver/internal/metadata"
 )
 
-type sqlServerScraper struct {
-	logger           *zap.Logger
-	config           *Config
-	watcherRecorders []watcherRecorder
-	mb               *metadata.MetricsBuilder
+type SQLServerScraperHelper struct {
+	id                 component.ID
+	sqlQuery           string
+	instanceName       string
+	scrapeCfg          scraperhelper.ControllerConfig
+	clientProviderFunc sqlquery.ClientProviderFunc
+	dbProviderFunc     sqlquery.DbProviderFunc
+	logger             *zap.Logger
+	telemetry          sqlquery.TelemetryConfig
+	client             sqlquery.DbClient
+	db                 *sql.DB
+	mb                 *metadata.MetricsBuilder
 }
 
-// watcherRecorder is a struct containing perf counter watcher along with corresponding value recorder.
-type watcherRecorder struct {
-	watcher  winperfcounters.PerfCounterWatcher
-	recorder recordFunc
+var _ scraperhelper.Scraper = (*SQLServerScraperHelper)(nil)
+
+func NewScraper(id component.ID,
```

Comment:
I was initially hoping to directly use `internal/sqlquery`'s scraper functionality, but it wouldn't have access to the sqlserver receiver's metadata.yaml, and metric emitting functionality. It simply returns a set of metrics, with no way for us to use sqlserver metadata on it, AFAIK.